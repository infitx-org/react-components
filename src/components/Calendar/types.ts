export type PossibleDate = Date | undefined;

export type PossibleDay = string | undefined;

export type Matrix = PossibleDay[][];

export type DisabledDays = (day: Date) => boolean;

export type DateRange = [PossibleDate, PossibleDate];
