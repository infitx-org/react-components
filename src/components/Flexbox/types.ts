export enum TopBottom {
  "top" = "flex-start",
  "bottom" = "flex-end",
}
export enum LeftRight {
  "left" = "flex-start",
  "right" = "flex-end",
}

export enum Mappers {
  "top" = "flex-start",
  "bottom" = "flex-end",
  "left" = "flex-start",
  "right" = "flex-end",
}

export type AlignItems =
  | "flex-start"
  | "flex-end"
  | "center"
  | "baseline"
  | "stretch";

export type JustifyContent =
  | "flex-start"
  | "flex-end"
  | "center"
  | "space-between"
  | "space-around"
  | "space-evenly";

export type AlignWithMap<T extends string> = `${AlignItems}` | `${T}`;
export type JustifyWithMap<T extends string> = `${JustifyContent}` | `${T}`;
