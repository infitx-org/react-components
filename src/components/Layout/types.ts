export enum AlignMap {
  "top" = "flex-start",
  "bottom" = "flex-end",
}

export enum JustifyMap {
  "left" = "flex-start",
  "right" = "flex-end",
}

export type AlignItems =
  | "flex-start"
  | "flex-end"
  | "center"
  | "baseline"
  | "stretch";

export type JustifyContent =
  | "flex-start"
  | "flex-end"
  | "center"
  | "space-between"
  | "space-around"
  | "space-evenly";

export type AlignWithMap = `${AlignItems | keyof typeof AlignMap}`;
export type JustifyWithMap = `${JustifyContent | keyof typeof JustifyMap}`;
