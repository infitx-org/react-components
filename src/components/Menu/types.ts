export interface MenuSectionProps {
  label: string;
  hidden?: boolean;
  icon?: React.ReactElement<React.SVGProps<SVGSVGElement>>;
  fill?: string;
  size?: number;
  disabled?: boolean;
  children: MenuItemElement | MenuItemElement[];
}

interface BaseMenuItemProps {
  label: string;
  active?: boolean;
  disabled?: boolean;
  hidden?: boolean;
  back?: boolean;
  icon?: React.ReactElement<React.SVGProps<SVGSVGElement>>;
  fill?: string;
  size?: number;
  // prop partial is not used directly by the MenuItem but it is used by the Menu component
  partial?: boolean;
  children?: MenuItemElement | MenuElement[];
}

interface BaseMenuItemPropsWithTo extends BaseMenuItemProps {
  to: string;
  path?: string;
}
interface BaseMenuItemPropsWithPath extends BaseMenuItemProps {
  to?: string;
  path: string;
}

export type MenuItemProps = BaseMenuItemPropsWithTo | BaseMenuItemPropsWithPath;

export type MenuItemElement = React.ReactElement<MenuItemProps>;
export type MenuSectionElement = React.ReactElement<MenuSectionProps>;
export type MenuElement = MenuItemElement | MenuSectionElement;
