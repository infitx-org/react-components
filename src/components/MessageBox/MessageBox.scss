@use "sass:map";
@import "../../scss/base/colors.scss";
@import "../../scss/base/shadows.scss";
@import "../../scss/themes.scss";
@import "../../scss/mixins.scss";

.rc-message-box {
  @include rc-component;
  border: 2px solid;
  min-height: 24px;
  padding: 3px 7px;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: initial;
  border-color: transparent;
  overflow: hidden;
}

.rc-message-box--default {
  fill: #333;
  color: #333;
  background-color: #eee;

  &.rc-message-box--inverted {
    fill: #eee;
    color: #eee;
    background-color: #333;
  }
}

@each $kind in $kinds {
  $base-color: map-get($theme-colors, $kind);
  $light-color: lighten(map-get($theme-shadows, $kind), 10%);

  .rc-message-box--#{$kind} {
    fill: $base-color;
    color: $base-color;
    background-color: $light-color;

    &.rc-message-box--inverted {
      fill: $light-color;
      color: $light-color;
      background-color: $base-color;
    }
  }
}

.rc-message-box--centered {
  align-items: center;
  justify-content: center;
}

.rc-message-box--active {
  border-color: currentColor;
}

.rc-message-box__icon-box {
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0px 2px;
}

.rc-message-box__messages {
  flex: 1 1 auto;
}
.rc-message-box__messages--centered {
  flex: 0 0 auto;
}
