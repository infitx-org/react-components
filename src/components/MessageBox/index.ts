export { default, MessageBoxProps } from "./MessageBox";
