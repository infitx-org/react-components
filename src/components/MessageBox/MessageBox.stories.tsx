import Row from "components/Flexbox/Row";
import { Kind } from "types";
import TestIcon from "resources/icons/test.svg";
import MessageBox from "./MessageBox";

export default {
  title: "Components/MessageBox",
  component: MessageBox,
};

const icon = <TestIcon />;

const Template = (args) => <MessageBox {...args} />;

export const Default = Template.bind({});
Default.args = {
  message: "I am a MessageBox",
  kind: "default",
  className: undefined,
  id: undefined,
  icon: undefined,
  fill: undefined,
  active: undefined,
};

export const MultipleMessages = Template.bind({});
MultipleMessages.args = {
  ...Default.args,
  message: ["First", "Second", "Third"],
};

export const Active = Template.bind({});
Active.args = {
  ...Default.args,
  active: true,
};

export const Centered = Template.bind({});
Centered.args = {
  ...Default.args,
  center: true,
};

export const WithIcon = Template.bind({});
WithIcon.args = {
  ...Default.args,
  icon,
};

export const WithIconFill = Template.bind({});
WithIconFill.args = {
  ...Default.args,
  icon,
  fill: "#f00",
};

export const WithId = Template.bind({});
WithId.args = {
  ...Default.args,
  id: "test-id",
};

const VariantsTemplate = (args) => (
  <Row align="space-between">
    <MessageBox message="default" kind="default" {...args} />
    {Object.values(Kind).map((kind) => {
      return <MessageBox key={kind} message={kind} kind={kind} {...args} />;
    })}
  </Row>
);
export const Variants = VariantsTemplate.bind({});
Variants.args = {};

export const VariantsInverted = VariantsTemplate.bind({});
VariantsInverted.args = {
  inverted: true,
};

export const VariantsWithIcon = VariantsTemplate.bind({});
VariantsWithIcon.args = {
  icon,
};

export const VariantsActive = VariantsTemplate.bind({});
VariantsActive.args = {
  active: true,
  icon,
};
