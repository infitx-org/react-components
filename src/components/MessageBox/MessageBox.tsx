import classnames from "classnames";
import { Kind } from "types";
import Icon from "components/Icon";
import "./MessageBox.scss";

function splitLines(prev: string[], curr: string): string[] {
  return [...prev, ...curr.split(`\n`)];
}

function getMessageComponent(
  message: string,
  index: number
): React.ReactElement {
  return (
    <div key={index} className="el-message-box__message">
      {message}
    </div>
  );
}

function getMessages(message: string | string[]): React.ReactElement[] {
  const subMessages = typeof message === "string" ? [message] : message;
  return subMessages.reduce(splitLines, []).map(getMessageComponent);
}

export interface MessageBoxProps {
  id?: string;
  className?: string;
  style?: React.CSSProperties;
  kind?: `${Kind}` | "default";
  icon?: React.ReactElement<React.SVGProps<SVGSVGElement>>;
  fill?: string;
  message?: string | string[];
  children?: React.ReactNode;
  size?: number;
  fontSize?: number;
  center?: boolean;
  active?: boolean;
  inverted?: boolean;
}

export default function MessageBox({
  id,
  className,
  style,
  kind = "default",
  icon,
  fill = "inherit",
  message,
  children,
  size = 20,
  fontSize = 13,
  center,
  active,
  inverted,
}: MessageBoxProps) {
  if (!message && !children) {
    return null;
  }

  const higherSize = Math.max.apply(Math, [icon ? size : 0, fontSize, 20]);

  const messageBoxStyle = {
    ...style,
    padding: `${higherSize / 2}px`,
    borderWidth: `${higherSize / 10}px`,
  };

  const iconStyle = { marginRight: `${higherSize / 2}px` };

  const messagesStyle = {
    fontSize: `${fontSize}px`,
  };

  const messageBoxClassName = classnames([
    "rc-message-box",
    `rc-message-box--${kind}`,
    active && `rc-message-box--active`,
    inverted && `rc-message-box--inverted`,
    center && "rc-message-box--centered",
    className,
  ]);

  const messagesClassName = classnames([
    "rc-message-box__messages",
    center && "rc-message-box__messages--centered",
  ]);

  return (
    <div className={messageBoxClassName} id={id} style={messageBoxStyle}>
      {icon && (
        <div className="rc-message-box__icon-box" style={iconStyle}>
          <Icon
            icon={icon}
            size={size}
            className="rc-message-box__icon"
            fill={fill}
          />
        </div>
      )}
      <div className={messagesClassName} style={messagesStyle}>
        {children || getMessages(message as string)}
      </div>
    </div>
  );
}
