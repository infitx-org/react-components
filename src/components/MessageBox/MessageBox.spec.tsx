import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import TestIcon from "resources/icons/test.svg";
import { Kind } from "types";
import MessageBox from "./MessageBox";

const icon = <TestIcon />;

describe("test the MessageBox component", () => {
  it("renders the component", () => {
    const { container } = render(<MessageBox message="message" />);
    expect(container.querySelector(".rc-message-box")).toBeTruthy();
  });

  it("renders null if no message or children are set", () => {
    const { container } = render(<MessageBox />);
    expect(container.querySelector(".rc-message-box")).toBeFalsy();
  });

  it("renders the message", () => {
    const { container } = render(<MessageBox message="message" />);
    expect(container.querySelector(".rc-message-box__messages")).toBeTruthy();
    expect(
      container.querySelector(".rc-message-box__messages")
    ).toHaveTextContent("message");
  });

  it("renders multiple messages", () => {
    const { container } = render(
      <MessageBox message={["message", "message2"]} />
    );
    expect(container.querySelector(".rc-message-box__messages")).toBeTruthy();
    expect(container.querySelectorAll(".rc-message-box__message").length).toBe(
      2
    );
  });

  it("renders the children before the message", () => {
    const { container } = render(
      <MessageBox message="message">
        <span>child</span>
      </MessageBox>
    );
    expect(container.querySelector(".rc-message-box__messages")).toBeTruthy();
    expect(
      container.querySelector(".rc-message-box__messages")
    ).toHaveTextContent("child");
  });

  it("renders the prop center", () => {
    const { container } = render(<MessageBox message="message" center />);
    expect(container.querySelector(".rc-message-box")).toHaveClass(
      "rc-message-box--centered"
    );
  });

  it("renders the prop active", () => {
    const { container } = render(<MessageBox message="message" active />);
    expect(container.querySelector(".rc-message-box")).toHaveClass(
      "rc-message-box--active"
    );
  });

  it("renders the prop inverted", () => {
    const { container } = render(<MessageBox message="message" inverted />);
    expect(container.querySelector(".rc-message-box")).toHaveClass(
      "rc-message-box--inverted"
    );
  });

  it("renders the icon", () => {
    const { container } = render(<MessageBox message="message" icon={icon} />);
    expect(container.querySelector(".rc-message-box__icon")).toBeTruthy();
  });

  it("renders the prop fill", () => {
    const { container } = render(
      <MessageBox message="message" icon={icon} fill="#fff" />
    );
    expect(container.querySelector("svg")).toHaveAttribute("fill", "#fff");
  });

  it("renders the prop className", () => {
    const { container } = render(
      <MessageBox message="message" className="test" />
    );
    expect(container.querySelector(".rc-message-box")).toHaveClass("test");
  });

  it("renders the prop id", () => {
    const { container } = render(
      <MessageBox message="message" id="testMessageBoxId" />
    );
    expect(container.querySelector(".rc-message-box")).toHaveAttribute(
      "id",
      "testMessageBoxId"
    );
  });

  it("renders the default correct kind", () => {
    const { container } = render(<MessageBox message="message" />);
    expect(container.querySelector(".rc-message-box")).toHaveClass(
      "rc-message-box--default"
    );
  });

  it("renders all the kinds", () => {
    Object.values(Kind).forEach((kind) => {
      const { container } = render(
        <MessageBox message="message" kind={kind} />
      );
      expect(container.querySelector(".rc-message-box")).toHaveClass(
        `rc-message-box--${kind}`
      );
    });
  });

  it("renders the correct font size", () => {
    const { container } = render(
      <MessageBox message="message" fontSize={40} />
    );
    expect(
      window.getComputedStyle(
        container.querySelector(".rc-message-box__messages") as Element
      ).fontSize
    ).toBe("40px");
  });

  it("renders the correct icon size", () => {
    const { container } = render(
      <MessageBox message="message" icon={icon} size={40} />
    );
    expect(container.querySelector("svg")).toHaveAttribute("width", "40");
    expect(container.querySelector("svg")).toHaveAttribute("height", "40");
  });

  it("renders the component correctly when multiple props are set", () => {
    const { container } = render(
      <MessageBox
        message="message"
        icon={icon}
        kind="secondary"
        active
        fill="#fff"
      />
    );
    expect(container).toMatchSnapshot();
  });
});
