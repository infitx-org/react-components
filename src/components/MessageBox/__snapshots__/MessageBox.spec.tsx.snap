// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the MessageBox component renders the component correctly when multiple props are set 1`] = `
<div>
  <div
    class="rc-message-box rc-message-box--secondary rc-message-box--active"
    style="padding: 10px; border-width: 2px;"
  >
    <div
      class="rc-message-box__icon-box"
      style="margin-right: 10px;"
    >
      <svg
        class="rc-message-box__icon"
        fill="#fff"
        height="20"
        stroke="transparent"
        width="20"
      />
    </div>
    <div
      class="rc-message-box__messages"
      style="font-size: 13px;"
    >
      <div
        class="rc-message-box__message"
      >
        message
      </div>
    </div>
  </div>
</div>
`;
